test/new-e2e/tests/sysprobe-functional/sysprobe_test.go

```diff
@@ -0,0 +1,131 @@
+	windowsAgent "github.com/DataDog/datadog-agent/test/new-e2e/tests/windows/common/agent"
+	componentsos "github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+	"github.com/stretchr/testify/require"
+)
+
+type vmSuite struct {
+	e2e.BaseSuite[environments.Host]
+
+	testspath string
+}
+
+var (
+	devMode = flag.Bool("devmode", false, "run tests in dev mode")
+)
+
+func TestVMSuite(t *testing.T) {
+	suiteParams := []e2e.SuiteOption{e2e.WithProvisioner(awshost.ProvisionerNoAgentNoFakeIntake(awshost.WithEC2InstanceOptions(ec2.WithOS(componentsos.WindowsDefault))))}
+	if *devMode {
+		suiteParams = append(suiteParams, e2e.WithDevMode())
+	}
+
+	e2e.Run(t, &vmSuite{}, suiteParams...)
+}
+
+func (v *vmSuite) SetupSuite() {
+	t := v.T()
+
+	// Get the absolute path to the test assets directory
+	currDir, err := os.Getwd()
+	require.NoError(t, err)
+
+	reporoot, _ := filepath.Abs(filepath.Join(currDir, "..", "..", "..", ".."))
+	kitchenDir := filepath.Join(reporoot, "test", "kitchen", "site-cookbooks")
+	v.testspath = filepath.Join(kitchenDir, "dd-system-probe-check", "files", "default", "tests")
+}
+
+func (v *vmSuite) TestSystemProbeSuite() {
+	v.BaseSuite.SetupSuite()
+	t := v.T()
+	// get the remote host
+	vm := v.Env().RemoteHost
+
+	err := windows.InstallIIS(vm)
+	require.NoError(t, err)
+	// HEADSUP the paths are windows, but this will execute in linux. So fix the paths
+	t.Log("IIS Installed, continuing")
+
+	t.Log("Creating sites")
+	// figure out where we're being executed from.  These paths should be in
+	// native path separators (i.e. not windows paths if executing in ci/on linux)
+
+	_, srcfile, _, ok := runtime.Caller(0)
+	require.True(t, ok)
+	exPath := filepath.Dir(srcfile)
+
+	sites := []windows.IISSiteDefinition{
+		{
+			Name:        "TestSite1",
+			BindingPort: "*:8081:",
+			AssetsDir:   path.Join(exPath, "assets"),
+		},
+		{
+			Name:        "TestSite2",
+			BindingPort: "*:8082:",
+			AssetsDir:   path.Join(exPath, "assets"),
+		},
+	}
+
+	t.Logf("AssetsDir: %s", sites[0].AssetsDir)
+	err = windows.CreateIISSite(vm, sites)
+	require.NoError(t, err)
+	t.Log("Sites created, continuing")
+
+	rs := windows.NewRemoteExecutable(vm, t, "testsuite.exe", v.testspath)
+	err = rs.FindTestPrograms()
+	require.NoError(t, err)
+
+	err = rs.CreateRemotePaths()
+	require.NoError(t, err)
+
+	err = rs.CopyFiles()
+	require.NoError(t, err)
+
+	// install the agent (just so we can get the driver(s) installed)
+	agentPackage, err := windowsAgent.GetPackageFromEnv()
+	require.NoError(t, err)
+	remoteMSIPath, err := windowsCommon.GetTemporaryFile(vm)
+	require.NoError(t, err)
+	t.Logf("Getting install package %s...", agentPackage.URL)
+	err = windowsCommon.PutOrDownloadFile(vm, agentPackage.URL, remoteMSIPath)
+	require.NoError(t, err)
+
+	err = windowsCommon.InstallMSI(vm, remoteMSIPath, "", "")
+	t.Log("Install complete")
+	require.NoError(t, err)
+
+	time.Sleep(30 * time.Second)
```

Comment:
Sleeps are expensive?
This, unfortunately, is to deal with a known problem in the agent where it reports it's stoppable when it's not.  So, unfortunately, a probe wouldn't help.  I'll be happy to change this when the agent problem is resolved.

Reply:
> Sleeps are expensive?
Sleeps are runner time we keep to literally sleep. At our scale, 30 seconds in one single test running in one single job per pipeline means 0,5 minutes * 7000 pipelines per month, which is 1 hour per month, sleeping.

This is for one single 30 seconds sleep in one single test in one single job per pipeline. Given the cost of Linux runners, that is 17$ per month. Again, one single sleep in one single test. 

Imagine having spread all over tests: it does not scale.

That is why, if possible, we should rather use retries for operations we cannot control. Would it be possible to stop and check that it stopped, and retry if it did not? My assumption is that most of the time it would take less than 30 seconds to stop safely.

More details at https://datadoghq.atlassian.net/wiki/spaces/ADX/pages/3492283102/Agent+tests+guidelines#Avoid-Sleep